dashboard/src/actions/repos.ts

```diff
@@ -49,10 +51,14 @@ export type AppReposAction = typeof allActions[number];
 
 export const deleteRepo = (name: string) => {
   return async (dispatch: Dispatch<IStoreState>) => {
+    const repo = await AppRepository.get(name);
     await AppRepository.delete(name);
     dispatch(requestRepos());
     const repos = await AppRepository.list();
     dispatch(receiveRepos(repos.items));
+    if (repo.spec.auth && repo.spec.auth.secretKeyRef.name) {
+      Secret.delete(repo.spec.auth.secretKeyRef.name, "kubeapps");
```

Comment:
Instead of managing the Secret being deleted, I think you can just set ownerReferences on the Secret object which will go and delete it when the parent gets deleted. We do this in the AppRepository controller for CronJobs and Jobs (https://github.com/kubeapps/kubeapps/blob/master/cmd/apprepository-controller/controller.go#L383). See https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/ also.